Warning in STK under clang

## Bug Report
@trilinos/\<PackageName\>

### Description

I'm seeing this in clang in EMPIRE.  We compile with -werror so this is a blocker.

```
In file included from /projects/empire/install-for-panzer/CLANG-3.9.0-RELEASE-SERIAL-static/include/stk_io/SidesetTranslator.hpp:37:
/projects/empire/install-for-panzer/CLANG-3.9.0-RELEASE-SERIAL-static/include/stk_mesh/baseImpl/EquivalentEntityBlocks.hpp:43:34: warning: class 'Entity' was previously declared as a struct [-Wmismatched-tags]
namespace stk { namespace mesh { class Entity; } }
                                 ^
/projects/empire/install-for-panzer/CLANG-3.9.0-RELEASE-SERIAL-static/include/stk_mesh/base/Entity.hpp:48:8: note: previous use is here
struct Entity
       ^
/projects/empire/install-for-panzer/CLANG-3.9.0-RELEASE-SERIAL-static/include/stk_mesh/baseImpl/EquivalentEntityBlocks.hpp:43:34: note: did you mean struct here?
namespace stk { namespace mesh { class Entity; } }
                                 ^~~~~
                                 struct
1 warning generated.
```

--- a/packages/stk/stk_mesh/stk_mesh/baseImpl/EquivalentEntityBlocks.hpp
+++ b/packages/stk/stk_mesh/stk_mesh/baseImpl/EquivalentEntityBlocks.hpp
@@ -40,7 +40,7 @@
 
 namespace stk { namespace mesh { class MetaData; } }
 namespace stk { namespace mesh { class BulkData; } }
-namespace stk { namespace mesh { class Entity; } }
+namespace stk { namespace mesh { struct Entity; } }
 namespace stk { namespace mesh { class Part; } }
 
 namespace stk {

